### PR TITLE
fix: add MaybePlatformPath

### LINKED ISSUE #2230: node --config is no longer configurable
### Describe the bug

since the argument has a default value it can be overridden via user input

ref https://github.com/paradigmxyz/reth/issues/2207

in fact, all of the path arguments suffer from this

### Steps to reproduce

- 

### Node logs

_No response_

### Platform(s)

_No response_

### What version/commit are you on?

_No response_

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by mattsse**
Closes #2230
ref #2207

respects user input again and only falls back to chain specific defaults if no user input.

Note: to bypass how clap `default_value` works, a `None` displays `default`

### CODE REVIEW

File: bin/reth/src/dirs.rs [hunk 1]
```diff
@@ -282,3 +337,30 @@ impl<D> From<ChainPath<D>> for PathBuf {
...
+
+    #[test]
+    fn test_maybe_config_platform_path() {
+        let path = MaybePlatformPath::<ConfigPath>::default();
+        let path = path.unwrap_or_chain_default(Chain::mainnet());
+        assert!(path.as_ref().ends_with("reth/mainnet/reth.toml"), "{:?}", path);
+
+        let path = MaybePlatformPath::<DbPath>::from_str("my/path/to/reth.toml").unwrap();
+        let path = path.unwrap_or_chain_default(Chain::mainnet());
+        assert!(path.as_ref().ends_with("my/path/to/reth.toml"), "{:?}", path);
```
Comment (Rjected) [inline bin/reth/src/dirs.rs:364]: nice, yeah going forward we should definitely unit test changes to dir structure cli flags

File: bin/reth/src/dirs.rs [hunk 2]
```diff
@@ -246,6 +252,55 @@ impl<D> PlatformPath<D> {
...
+}
+
+impl<D> FromStr for MaybePlatformPath<D> {
+    type Err = shellexpand::LookupError<VarError>;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let p = match s {
+            "default" => {
+                // NOTE: this is a workaround for making it work with clap's `default_value_t` which
+                // computes the default value by via `Default -> Display -> FromStr`
```
Comment (Rjected) [inline bin/reth/src/dirs.rs:295]: ah I see, yeah cursed

**Review by Rjected — APPROVED** [review]
Nice! really like the added tests

**Review by gakonst — APPROVED** [review]

**Comment by gakonst** [comment]
oops approved previous commits but overall lgtm

**Pull request merged**